libyul/backends/evm/AsmCodeGen.cpp

```diff
@@ -146,22 +146,32 @@ pair<shared_ptr<AbstractAssembly>, AbstractAssembly::SubID> EthAssemblyAdapter::
 	return {make_shared<EthAssemblyAdapter>(*assembly), static_cast<size_t>(sub.data())};
 }
 
-void EthAssemblyAdapter::appendDataOffset(AbstractAssembly::SubID _sub)
+void EthAssemblyAdapter::appendDataOffset(std::vector<AbstractAssembly::SubID> const& _subIds)
 {
-	auto it = m_dataHashBySubId.find(_sub);
-	if (it == m_dataHashBySubId.end())
-		m_assembly.pushSubroutineOffset(_sub);
-	else
-		m_assembly << evmasm::AssemblyItem(evmasm::PushData, it->second);
+	yulAssert(_subIds.size() < 32, "");
+	if (_subIds.size() == 1)
+		if (auto it = m_dataHashBySubId.find(_subIds[0]); it != m_dataHashBySubId.end())
+		{
+			m_assembly << evmasm::AssemblyItem(evmasm::PushData, it->second);
+			return;
+		}
+
+	u256 encodedSubId = evmasm::Assembly::encodeSubIds(_subIds);
+	m_assembly.pushSubroutineOffset(encodedSubId);
```

Comment:
This is simple enough on a single line:
```suggestion
	m_assembly.pushSubroutineOffset(evmasm::Assembly::encodeSubIds(_subIds));
```